jax/experimental/mosaic/gpu/profiler.py

```diff
@@ -109,6 +98,90 @@ def run(*args, **kwargs):
   return outs, float(elapsed)
 
 
+def _measure_cupti(f, aggregate):
+  def wrapper(*args, **kwargs):
+    mosaic_gpu_lib._mosaic_gpu_ext._cupti_init()
+    try:
+      results = jax.block_until_ready(jax.jit(f)(*args, **kwargs))
```

Comment:
Shall we jit `f` in the enclosing namespace?

Reply:
Hmm what do you mean?

Reply:
Sorry, I meant

```
jit_f = jax.jit(f)
```

just before the definition of `wrapper`.

Reply:
Why? If you are thinking about performance, `jax.jit` doesn't actually JIT until the invocation anyway, right? And if the function has been compiled for the shapes and types before, then it's a quick cache look up anyway.

What am I missing?